iavl/iavl_tree.go

```diff
@@ -196,6 +198,16 @@ func (t *IAVLTree) GetRangeWithProof(startKey []byte, endKey []byte, limit int)
 	return t.getRangeWithProof(startKey, endKey, limit)
 }
 
+// GetFirstInRangeWithProof gets the first key/value pair in the specified range, with a proof.
+func (t *IAVLTree) GetFirstInRangeWithProof(startKey, endKey []byte) ([]byte, []byte, *KeyFirstInRangeProof, error) {
```

Comment:
Does it make sense to combine these two into one interface as you did with GetWithProof?